dbms/src/Storages/KVStore/MultiRaft/ApplySnapshot.cpp

```diff
@@ -80,8 +80,8 @@ void KVStore::checkAndApplyPreHandledSnapshot(const RegionPtrWrap & new_region,
             // engine may delete data unsafely.
             auto region_lock = region_manager.genRegionTaskLock(old_region->id());
             old_region->setStateApplying();
-            tmt.getRegionTable().tryWriteBlockByRegion(old_region);
-            tryFlushRegionCacheInStorage(tmt, *old_region, log);
+            // It is not worthy to call `tryWriteBlockByRegion` and `tryFlushRegionCacheInStorage` here,
+            // even if the written data is useful, it could be overwritten later in `onSnapshot`.
```

Comment:
```suggestion
            // even if the written data is useful, it could be overwritten later in `onSnapshot`.
            // Note that we must persistRegion. This is to ensure even if a restart happens before
            // the apply snapshot is finished, TiFlash can correctly reject the read index requests
```